src/lib/events/events.ts

```diff
@@ -0,0 +1,61 @@
+import { AxiosInstance } from 'axios'
+import { TimelyAppConfig, TimelyBulkUpdateEventsReturn, TimelyEvent } from '../types'
+
+export class Events {
+    constructor(private readonly http: AxiosInstance, private readonly config: TimelyAppConfig) {}
+
+    async getAll(start?: string, end?: string): Promise<TimelyEvent[]> {
+        return this.http.post(`/${this.config.accountId}/reports/filter.json`, {
+            // TODO: Are all properties required?
+            project_ids: '',
+            user_ids: '',
+            team_ids: '',
+            label_ids: '',
+            since: start ?? null,
+            until: end ?? null,
+            scope: 'events',
+            limit: 9999,
+            page: 1,
+            select: 'all',
+            sort: 'day',
+            order: 'desc',
+            published: true,
+        })
+    }
+
+    async getByProjectId(projectId: number, start: string, end: string): Promise<TimelyEvent[]> {
+        return this.http.get(
+            `/${this.config.accountId}/projects/${projectId}/events${
+                start ? `?since=${start}` : ''
+            }${end ? `&upto=${end}` : ''}`,
+        )
+    }
+
+    /**
+     *
+     * @param updateArray
+     *  example minimum keys for updateArray
+     * [
+     *   { id: 123456760, billed: true},
+     *   { id: 123456761, billed: true, label_ids: []},
+     *   { id: 123456762, billed: true, label_ids: [], project_id: 1234567 },
+     *   { id: 123456763, billed: true, label_ids: [], project_id: 1234567 },
+     *   { id: 123456764, billed: true, label_ids: [], project_id: 1234567 },
+     * ]
+     */
+    async bulkUpdate(updateArray: any[]): Promise<TimelyBulkUpdateEventsReturn[]> {
```

Comment:
Not a blocker, but @santese we should add a type here

Reply:
@CalebAlbers , I can rename the interface to more of a shared name, the Bulk Event Actions like Delete and Update,  have the same return

Reply:
Actually I'm blind , just realized what you pointed out , yeah let me add a type for that